src/main/scala/ch/epfl/bluebrain/nexus/kg/KgError.scala

```diff
@@ -0,0 +1,97 @@
+package ch.epfl.bluebrain.nexus.kg
+
+import akka.http.scaladsl.model.StatusCodes
+import cats.syntax.show._
+import ch.epfl.bluebrain.nexus.kg.config.Contexts.errorCtxUri
+import ch.epfl.bluebrain.nexus.kg.resources.{ProjectLabel, Ref}
+import ch.epfl.bluebrain.nexus.rdf.syntax.circe.context._
+import ch.epfl.bluebrain.nexus.service.http.directives.StatusFrom
+import io.circe.generic.extras.Configuration
+import io.circe.generic.extras.semiauto.deriveEncoder
+import io.circe.{Encoder, Json}
+
+/**
+  * Enumeration of runtime errors.
+  *
+  * @param msg a description of the error
+  */
+@SuppressWarnings(Array("IncorrectlyNamedExceptions"))
+sealed abstract class KgError(val msg: String) extends Exception with Product with Serializable {
+  override def fillInStackTrace(): KgError = this
+  override def getMessage: String          = msg
+}
+
+@SuppressWarnings(Array("IncorrectlyNamedExceptions"))
+object KgError {
+
+  /**
+    * Generic wrapper for kg errors that should not be exposed to clients.
+    *
+    * @param reason the underlying error reason
+    */
+  final case class InternalError(reason: String) extends KgError(reason)
+
+  /**
+    * Signals that the requested resource was not found
+    */
+  final case class NotFound(ref: Option[String] = None) extends KgError("The requested resource could not be found.")
```

Comment:
It would be useful to have the `ref` somewhere in the error message.

Reply:
what is the use case of having an `Option` here? when would it be `None`?

Reply:
the ref is added a a separate field, and the case for None happens when the the routes are terminated without knowing the exact path